IBAnimatable/BorderType.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  BorderType.swift
+//  IBAnimatable
+//
+//  Created by Tom Baranes on 13/01/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+
+public enum BorderType: IBEnum {
+  case solid
+  case dash(dashLength: Int, spaceLength: Int)
+  case none
+}
+
+extension BorderType {
+
+  public init(string: String?) {
+    guard let string = string else {
+      self = .none
+      return
+    }
+
+    let (name, params) = AnimationType.extractNameAndParams(from: string)
+    switch name {
+    case "solid":
+      self = .solid
+    case "dash":
+      self = .dash(dashLength: params[safe: 0]?.toInt() ?? 1, spaceLength: params[safe: 1]?.toInt() ?? 1)
+    default:
+      self = .none
+    }
+  }
+}
+
+extension BorderType: Equatable {
+}
+
+public func == (lhs: BorderType, rhs: BorderType) -> Bool {
```

Comment:
We need it for that line: `if borderType == .solid, borderSides == .AllSides`. That's the condition we used to check if we should use a custom layer or not.